mesonbuild/wrap/wrap.py

```diff
@@ -92,30 +131,37 @@ def open_wrapdburl(urlstring: str, allow_insecure: bool = False, have_opt: bool
         mlog.warning(f'SSL module not available in {sys.executable}: WrapDB traffic not authenticated.', once=True)
 
     # If we got this far, allow_insecure was manually passed
-    nossl_url = url._replace(scheme='http')
+    if has_ssl:
+        ctx = ssl.create_default_context()
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
+    else:
+        ctx = None
+        # e.g. https:// -> http://
+        urlstring_ = urlstring_.replace('s://', '://', 1)
```

Comment:
I don't like this replacement of the scheme. IMO it would be better to ask user to manually set insecure (http or ftp) wrapdb source if https/ftps are not working for whatever reason.

Reply:
If a non-standard port is used for the protocol this replacement will not work

Reply:
Replaced with an exception